manual/src/refman/patterns.etex

```diff
@@ -80,6 +106,13 @@ The pattern @pattern_1 "as" value-name@ matches the same values as
 the name @value-name@ is bound to the matched value, in addition to the
 bindings performed by the matching against @pattern_1@.
 
+\begin{caml_example}{toplevel}
+let option_exception f x =
+  try Some (f x) with
+  | Assert_failure _ as e -> raise e
+  | _ -> None;;
+\end{caml_example}
```

Comment:
As discussed over email, I would suggest the following, simpler example for `as` patterns:
```
let sort_pair ((x, y) as p) = if x <= y then p else (y, x)
```